### PR TITLE
add user_patch action, allow sysadmins to be managed via the web frontend

### PR DESCRIPTION
**Pull request by chris48s**
At the moment it is only possible to manage sysadmins via the CLI

This PR adds a `user_patch` action and permits a sysadmin to promote other users to sysadmin or revoke sysadmin permissions from a user without needing console access.

### Features:

- [x] includes tests covering changes
- [x] includes updated documentation (yes: docstring for action function)
- [x] includes user-visible changes
- [x] includes API changes
- [ ] includes bugfix for possible backport

### CODE REVIEW

**Comment by wardi** [comment]
Instead of `user_update_sysadmin` please add a `user_patch` action that is simpler and a more generic way of updating individual user fields

File: ckan/views/user.py [hunk 1]
```diff
@@ -815,6 +815,37 @@ def followers(id):
...
+        context = {
+            u'model': model,
+            u'session': model.Session,
+            u'user': g.user,
+            u'auth_user_obj': g.userobj,
+        }
+        data_dict = {u'id': id_, u'sysadmin': status}
+        user = logic.get_action(u'user_patch')(context, data_dict)
+    except logic.NotAuthorized:
+        return base.abort(403, u'Not authorized to promote user to sysadmin')
```
Comment (wardi) [inline ckan/views/user.py:832]: please use `_()` around these abort messages so they can be localized

File: ckan/templates/admin/index.html [hunk 2]
```diff
@@ -1,11 +1,77 @@
 {% extends "admin/base.html" %}
 
 {% block primary_content_inner %}
-  <ul class="user-list">
-    {% for user in sysadmins %}
-      <li>{{ h.linked_user(user) }}</li>
-    {% endfor %}
-  </ul>
+  <h2>Current Sysadmins</h2>
```
Comment (wardi) [inline ckan/templates/admin/index.html:4]: these strings need gettext `{{ _("...") }}` around them

File: ckan/templates/admin/index.html [hunk 3]
```diff
@@ -1,11 +1,77 @@
...
+          <td>{{ h.linked_user(user.name) }}</td>
+          <td>
+            <div class="btn-group pull-right">
+              <form method="POST" action="{% url_for 'user.sysadmin' %}">
+                <input type="hidden" value="{{ user.id }}" name="id" />
+                <input type="hidden" value="0" name="status" />
+                <button
+                  type="submit"
+                  class="btn btn-danger btn-sm"
+                  title="Revoke Sysadmin permission"
```
Comment (wardi) [inline ckan/templates/admin/index.html:25]: and here

File: ckan/templates/admin/index.html [hunk 4]
```diff
@@ -1,11 +1,77 @@
...
+            </div>
+          </td>
+        </tr>
+      {% endfor %}
+    </tbody>
+  </table>
+
+  <hr />
+
+  <h2>Promote user to Sysadmin</h2>
```
Comment (wardi) [inline ckan/templates/admin/index.html:39]: and here

File: ckan/templates/admin/index.html [hunk 5]
```diff
@@ -1,11 +1,77 @@
...
+            {% endfor %}
+          </select>
+          <input type="hidden" value="1" name="status" />
+        </div>
+
+        <div class="form-actions">
+          <button
+            type="submit"
+            class="btn btn-primary"
+            title="Promote user to Sysadmin"
```
Comment (wardi) [inline ckan/templates/admin/index.html:59]: and here

File: ckan/views/admin.py [hunk 6]
```diff
@@ -60,8 +60,12 @@ def before_request():
 
 
 def index():
-    data = dict(sysadmins=[a.name for a in _get_sysadmins()])
-    return base.render(u'admin/index.html', extra_vars=data)
+    context = {u'user': g.user}
+    users = logic.get_action(u'user_list')(context, {})
```
Comment (amercader) [inline ckan/views/admin.py:64]: This is not very efficient and might cause performance issues with many users. I suggest:

1. Getting sysadmin users with a small helper function that queries de model directly

2. Replacing the UI widget with the autocomplete one that is used in the add member or add collaborator forms

https://github.com/ckan/ckan/blob/master/ckan/templates/group/member_new.html#L31

**Pull request merged**